### PR TITLE
bugfix GFM-119 - use hof version of eslint and eslint-config-homeoffice

### PR DESCRIPTION
**Pull request by joefitter**
* removed from deps
* fixed lint

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -61,7 +59,7 @@ function secureCookies(req, res, next) {
   const cookie = res.cookie.bind(res);
   res.cookie = (name, value, options) => {
     options = options || {};
-    options.secure = (req.protocol === 'https');
+    options.secure = req.protocol === 'https';
```
Comment (easternbloc) [inline app.js]: not sure why you're hating on braces. IMHO it reads better with them.

File: app.js [hunk 2]
```diff
@@ -74,7 +72,7 @@ app.use(secureCookies);
 app.use(session({
   store: redisStore,
   cookie: {
-    secure: (config.env === 'development' || config.env === 'docker-compose' || config.env === 'ci') ? false : true
+    secure: !(config.env === 'development' || config.env === 'docker-compose' || config.env === 'ci')
```
Comment (easternbloc) [inline app.js]: The turnery was intentional to make it clear for future devs.

File: app.js (hunk 1 above)
Comment (joefitter) [inline app.js]: eslint says NO ;)

File: assets/js/index.js [hunk 3]
```diff
@@ -53,24 +50,24 @@ $('.typeahead').each(function applyTypeahead() {
       sorter: function sorter(a, b) {
         var input = $input.val();
         var startsWithInput = function startsWithInput(x) {
-          return (x.toLowerCase().substr(0, input.length) === input.toLowerCase()) ? -1 : 1;
+          return x.toLowerCase().substr(0, input.length) === input.toLowerCase() ? -1 : 1;
```
Comment (easternbloc) [inline assets/js/index.js]: haters

File: assets/js/index.js [hunk 4]
```diff
@@ -53,24 +50,24 @@ $('.typeahead').each(function applyTypeahead() {
...
         };
 
         var compareAlpha = function compareAlpha(x, y) {
-          var less = (x < y) ? -1 : 1;
-          return (x === y) ? 0 : less;
+          var less = x < y ? -1 : 1;
+          return x === y ? 0 : less;
```
Comment (easternbloc) [inline assets/js/index.js]: gonna

File: assets/js/index.js [hunk 5]
```diff
@@ -53,24 +50,24 @@ $('.typeahead').each(function applyTypeahead() {
...
         };
 
         var compareStartsWithInput = function compareStartsWithInput(x, y) {
           var startsWithFirst = startsWithInput(x);
           var startsWithSecond = startsWithInput(y);
 
-          return (startsWithFirst === startsWithSecond) ? 0 : startsWithFirst;
+          return startsWithFirst === startsWithSecond ? 0 : startsWithFirst;
```
Comment (easternbloc) [inline assets/js/index.js]: hate

File: app.js (hunk 1 above)
Comment (easternbloc) [inline app.js]: pff who even wants eslint :)

File: app.js (hunk 2 above)
Comment (joefitter) [inline app.js]: in that case the parenthesis should do as superfluous

File: assets/js/index.js (hunk 3 above)
Comment (joefitter) [inline assets/js/index.js]: superfluous 😛

File: assets/js/index.js (hunk 4 above)
Comment (joefitter) [inline assets/js/index.js]: lulz

File: assets/js/index.js (hunk 5 above)
Comment (joefitter) [inline assets/js/index.js]: don't you mean (haters (gunna) (h)(a)(t))(e)?

Comment (joefitter) [inline assets/js/index.js]: If I wanted to look at brackets all day I'd write lisp

File: app.js (hunk 1 above)
Comment (joefitter) [inline app.js]: I totally agree - I've said it before and I'll say it again - https://github.com/feross/standard

Comment (joefitter) [inline app.js]: git rid of those pesky semi-colons

Comment (joefitter) [inline app.js]: http://mislav.net/2010/05/semicolons/

**Comment by joefitter** [comment]
@gxxm @easternbloc @JoeChapman @bendiggle Mind giving this the 👍 ? How do we feel about extra brackets? I personally don't like them but if I am overruled then I guess we could put them in the `.eslintrc`

**Comment by easternbloc** [comment]
Personally over the years I've started to code for the next person and I think that longer code can often be more readable for the developer who has to pick this up when we're gone. I've tended to avoid shorthand in most cases so I'll add braces to immediately clarify the expression. BUT I get it can be overly verbose at times. It's your codebase after all so maybe this should be pushed towards @gxxm @bendiggle and @JoeChapman

**Comment by gxxm** [comment]
:thumbsup: :thumbsup:

**Comment by JoeChapman** [comment]
Sorry @easternbloc, I agree there are many times when verbosity can === readability, but I prefer to use fewer parentheses in the instances above. So this LGTM.

**Comment by allandegnan** [comment]
Not that it's a devops problem, but it'd be generally nice if you nodejs guys signed up to a documented standard and go forward adhering to that.  Personally I generally agree with Ben but that doesn't matter in the face being able to hire a guy who uses / knows foo standard.

**Comment by easternbloc** [comment]
Thanks @allandegnan we do have a standard but it's "maturing" and instances like this are adding/subtracting towards it.

**Comment by joefitter** [comment]
Thats why we have .eslint and .jscs - they are our documented standard - this repo wasn't using the global home office version

**Comment by joefitter** [comment]
on that note - @easternbloc https://github.com/UKHomeOffice/eslint-config-homeoffice/pull/13

File: app.js (hunk 2 above)
Comment (JoeChapman) [inline app.js]: Actually, I don't like this. I prefer the ternary

**Comment by allandegnan** [comment]
fair enough @easternbloc @joefitter; although I was slightly leaning more towards something like: https://www.python.org/dev/peps/pep-0008/ or http://www.php-fig.org/psr/psr-2/ rather than having an internally defined one.  I do understand that obviously these things aren't neccessaily always complete or well documented in any one particularl language.

**Comment by joefitter** [comment]
@allandegnan YES - https://github.com/feross/standard

**Comment by joefitter** [comment]
I have been saying to use that since I started, but @easternbloc loves his semicolons too much

**Comment by easternbloc** [comment]
I literally can't get enough of `;`

**Comment by joefitter** [comment]
[I love ;](javascript: document.body.innerText = document.body.innerText.split('').join(';'))

**Comment by joefitter** [comment]
boo - it stripped my bookmarklet :(

**Pull request merged**